kong/db/schema/entities/filter_chains.lua

```diff
@@ -18,22 +18,41 @@ local wasm = require "kong.runloop.wasm"
 
 ---@class kong.db.schema.entities.wasm_filter : table
 ---
----@field name    string
----@field enabled boolean
----@field config  string|table|nil
+---@field name        string
+---@field enabled     boolean
+---@field config      string|nil
+---@field json_config any|nil
 
 
 local filter = {
   type = "record",
   fields = {
-    { name    = { type = "string", required = true, one_of = wasm.filter_names,
-                  err = "no such filter", }, },
-    { config  = { type = "string", required = false, }, },
-    { enabled = { type = "boolean", default = true, required = true, }, },
+    { name       = { type = "string", required = true, one_of = wasm.filter_names,
+                     err = "no such filter", }, },
+    { config     = { type = "string", required = false, }, },
+    { enabled    = { type = "boolean", default = true, required = true, }, },
+
+    { json_config = {
+        type = "json",
+        required = false,
+        json_schema = {
+          parent_subschema_key = "name",
+          namespace = constants.SCHEMA_NAMESPACES.PROXY_WASM_FILTERS,
+          optional = true,
+        },
+      },
+    },
+
+  },
+  entity_checks = {
```

Comment:
Is there a reason we don't need `at_least_one_of` for config, json_config?

Reply:
There can be filters that take no config.

Reply:
Yeah, configuration needs to remain optional, as it's perfectly valid to have a proxy-wasm filter that requires no configuration to function.

Specifying a schema for your filter's `json_config` using this new `my-filter.meta.json` method is one means of making it effectively required though.